terminal total difficulty override setting (#2587)
ethereum/consensus-specs: specs/merge/client_settings.md

# The Merge -- Client Settings

**Notice**: This document is a work-in-progress for researchers and implementers.

> This document specifies configurable settings that merge clients are expected to ship with.

Reviewer: ```suggestion
This document specifies configurable settings that clients must implement for the Merge.
```

### Override terminal total difficulty

> To coordinate changes to [`terminal_total_difficulty`](fork-choice.md#transitionstore), clients

Reviewer: ```suggestion
To coordinate manual overrides to [`terminal_total_difficulty`](fork-choice.md#transitionstore), clients
```

> should have a setting `--terminal-total-difficulty-override`.

Reviewer: ```suggestion
must provide `--terminal-total-difficulty-override` as a configurable setting.
```

> If `TransitionStore` has already been initialized, this just changes the value of

Reviewer 2: I would also add that the `engine_terminalTotalDifficultyOverride` message should be sent to the execution client.

Reviewer: I might instead specify it wrt the EIP 3675 events rather than the API. This leaves it generic enough for alternative merge-client implementations (e.g. a unified client that doesn't  utilize an API for comms on the layer

Reviewer: ```suggestion
If `TransitionStore` has already [been initialized](./fork.md#initializing-transition-store), this alters the previously initialized value of
```

Author: > I might instead specify it wrt the EIP 3675 events rather than the API. This leaves it generic enough for alternative merge-client implementations (e.g. a unified client that doesn't utilize an API for comms on the layer

agree with this

> `TransitionStore.terminal_total_difficulty`, otherwise it initializes `TransitionStore` with the specified

Reviewer: ```suggestion
`TransitionStore.terminal_total_difficulty`, otherwise this setting initializes `TransitionStore` with the specified, bypassing `compute_terminal_total_difficulty` and the use of an `anchor_pow_block`.
```

`terminal_total_difficulty`.

> By default, this setting is expected to not be used and `terminal_total_difficulty` will be set as defined

Reviewer: ```suggestion
Except under exceptional scenarios, this setting is expected to not be used, and `terminal_total_difficulty` will operate with [default functionality](./fork.md#initializing-transition-store). Sufficient warning to the user about this exceptional configurable setting should be provided.
```
